src/megaapi_impl.cpp

```diff
@@ -7336,7 +7345,7 @@ void MegaApiImpl::logout_result(error e)
     MegaRequestPrivate* request = requestMap.at(client->restag);
     if(!request || (request->getType() != MegaRequest::TYPE_LOGOUT)) return;
 
-    if(!e)
+    if(!e || e == API_ESID)
```

Comment:
Since the command doesn't receive the ESID and the SDK doesn't call `logout_result()` with that error, I agree with you.